Azure tutorials (#104)
streamsets/tutorials: working-with-azure/hdinsightkafka_to_sqldw_and_blobstorage.md

> # Ingesting Data from HDInsight Kafka into Azure SQL Datawarehouse and Hive backed by Blob Storage

Reviewer: ```suggestion
# Ingesting Data from Apache Kafka on HDInsight into Azure SQL Data Warehouse and Apache Hive backed by Azure Blob Storage
```

> [Kafka](https://docs.microsoft.com/en-us/azure/hdinsight/kafka/apache-kafka-introduction) is an open-source, distributed streaming platform. It's often used as a message broker, as it provides functionality similar to a publish-subscribe message queue. [HDInsight Kafka](https://azure.microsoft.com/en-us/services/hdinsight/apache-kafka/) is a managed service that provides a simplified configuration for Apache Kafka.

Reviewer: ```suggestion
[Kafka](https://docs.microsoft.com/en-us/azure/hdinsight/kafka/apache-kafka-introduction) is an open-source, distributed streaming platform. It's often used as a message broker, as it provides functionality similar to a publish-subscribe message queue. [Apache Kafka on HDInsight](https://azure.microsoft.com/en-us/services/hdinsight/apache-kafka/) is a managed service that provides a simplified configuration for Apache Kafka.
```

> [SQL Data Warehouse](https://docs.microsoft.com/en-us/azure/sql-data-warehouse/sql-data-warehouse-overview-what-is) is a cloud-based Enterprise Data Warehouse (EDW) that leverages Massively Parallel Processing (MPP) to quickly run complex queries across petabytes of data. Use SQL Data Warehouse as a key component of a big data solution. SQL Data Warehouse stores data into relational tables with columnar storage. Once data is stored in SQL Data Warehouse, you can run analytics at massive scale.

Reviewer: ```suggestion
[Azure SQL Data Warehouse](https://docs.microsoft.com/en-us/azure/sql-data-warehouse/sql-data-warehouse-overview-what-is) is a cloud-based Enterprise Data Warehouse (EDW) that leverages Massively Parallel Processing (MPP) to quickly run complex queries across petabytes of data. Use SQL Data Warehouse as a key component of a big data solution. SQL Data Warehouse stores data into relational tables with columnar storage. Once data is stored in SQL Data Warehouse, you can run analytics at massive scale.
```

[Azure Blob Storage](https://azure.microsoft.com/en-us/services/storage/blobs/ "Azure Blob Storage") can handle all your unstructured data, scaling up or down as required. You no longer have to manage it, you only pay for what you use, and you save money over on-premises storage options.

## Goal

> In this tutorial, you will learn how to leverage Azure services with StreamSets to read data from an HDInsight Kafka stream into ingest into Azure SQL Datawarehouse and a Hive table backed by Blob Storage.

Reviewer: ```suggestion
In this tutorial, you will learn how to leverage Azure services with StreamSets to read data from an Apache Kafka on HDInsight stream into ingest into Azure SQL Datawarehouse and a Hive table backed by Blob Storage.
```

## Prerequisites

> ## Creat the SQL Datawarehouse table

Reviewer: ```suggestion
## Creat the SQL Data Warehouse table
```

When the Hive Metadata processor detects a schema change, it sets the roll header attribute to signal to the destination that the data file should be ‘rolled’ - that is, the current file closed and a new file opened.

We set **Idle Timeout** set to `10 SECONDS` so the destination closes the file within 10 seconds of writing the data, since we want to see data immediately. If we left the defaults in place, we might not see some data in Hive until an hour after it was written. This might be appropriate for a production deployment, but would make a very time-consuming tutorial!

To complete the pipeline, add a Hive Metastore destination and select the Stage Library based on your HDP version. Connected the input of this destination to the Hive Metadata processor’s #2 output and configure Hive settings like this:

![image alt text](img/KafkaToSqlDWandHive/HiveMetastore_Hive.png)

Again, Validate and Preview the pipeline to ensure all is well.

## Running the Pipeline

If all is well, it’s time to run the pipeline! Hit the run button ![image alt text](img/BlobToKafka/PlayIcon.png) and you should see 5386 input records and 5386 output records in the monitoring panel.

## Conclusion

> This tutorial shows how simple it is to stream data from any Kafka into multiple destinations with Azure Services. For Realtime trickle feeds, you can directly send data into SQL Datawarehouse and for Archival, you can send data into Hive.

Reviewer: ```suggestion
This tutorial shows how simple it is to stream data from any Kafka topic into multiple destinations with Azure Services. For real-time trickle feeds, you can directly send data into Azure SQL Data Warehouse and for archival, you can send data into Hive.
```
